ransom-note.md

```diff
@@ -0,0 +1,27 @@
+# Thread Template
+
+Hi friends! The REPLACE WITH DAY OF WEEK problem is [383. Ransom Note](https://leetcode.com/problems/ransom-note/). We have live discussions at **8-8:15am**, **7-7:15pm**, and **9:30-9:45pm** (all times California time) for anyone who wants to discuss or see the problem solved live.
+
+* **Baseline goal**: Get accepted in any way at all! Hint: ||one way to solve it would be for each character in "abcdefghijklmnopqrstuvwxyz" count whether it occurs more times in the magazine than in the ransom note||.
+* **Stretch goal**: Iterate only once over `ransomNote` and only once over `magazine`. If your language has built-in libraries for character counting try a version of the solution that doesn't use them.
+
+I look forward to seeing your solutions! 
+
+Some reference solutions for inspiration:
+* O(alphabet size) iterations ([TypeScript](https://leetcode.com/problems/ransom-note/submissions/1048453389/), [Python](https://leetcode.com/problems/ransom-note/submissions/1048446556/), [Java](https://leetcode.com/problems/ransom-note/submissions/1048451774/), [Ruby](https://leetcode.com/problems/ransom-note/submissions/1048458944/))
+* one iteration only ||using frequency map|| ([TypeScript](https://leetcode.com/problems/ransom-note/submissions/1021115482/), [Python](https://leetcode.com/problems/ransom-note/submissions/1048445154/), [Java](https://leetcode.com/problems/ransom-note/submissions/1048449200/), [Ruby](https://leetcode.com/problems/ransom-note/submissions/1048459505/), [C](https://leetcode.com/problems/ransom-note/submissions/1048456674/))
+* with some help from built-ins ([Python](https://leetcode.com/problems/ransom-note/submissions/1048442249/), [Ruby](https://leetcode.com/problems/ransom-note/submissions/1048443409/))
+
+# Notes for Facilitators
+
+## Checklist for a good session:
+
+* controlled the clock
+  * started on time
+  * ended on time
+* fostered a positive learning environment
+  * led the discussion with enthusiasm
+  * involved the audience
+  * confirmed that people understand the code being shown
+
```

Comment:
- cover the solution thats uses only a single array to store the occurrences of each character
- cover the solution using frequency map, only using one iteration. If time permits cover solution in other languages, preferably TypeScript/Python.
- if time permits, cover the solutions using some kind of built-in function.

Reply:
> cover the solution thats uses only a single array to store the occurrences of each character

Do you mean the solution that uses an array of size 26? That seems like a nice bonus optimization. I wouldn't consider that required for the "good" solution.

Reply:
> If time permits cover solution in other languages, preferably TypeScript/Python.

That implies that you didn't start with TypeScript or Python to begin with 😛